Board.cpp

```diff
@@ -100,8 +102,25 @@ bool Board::makeMove(Move m)
 		}
 		else if (m.m.mtype & 16)
 		{
-			// pawn promotion
-			// TODO
+			uint8_t piece = m.m.detail;
+			PType promotionPiece;
+				if(piece & 1)
+					promotionPiece = knight;
+				else if(piece & 2)
+					promotionPiece = bishop;
+				else if(piece & 4)
+					promotionPiece = rook;
+				else if(piece & 8)
+					promotionPiece = queen;
+				else {
+					cout << "Error no detail value for pawn promotion\n";
```

Comment:
I think it would be better if we avoid side effects (writing to stdout) in the Board member functions. Maybe we could add logging to stderr instead just to keep the streams separate? Because rn we use the program on the command line so it could get confusing if we add printing here. I think we just return false and allow whatever function is higher up to make a decision.

Reply:
Oh yea, this was leftover from when I had it as a switch, I am just going to delete it for now, since the return false shows the user a print. But really it should throw an error so we have to deal with that eventually.